docs/standard/collections/index.md

```diff
@@ -11,78 +11,101 @@ helpviewer_keywords:
+|A sorted collection|<xref:System.Collections.Generic.SortedList%602>|<xref:System.Collections.SortedList>|<xref:System.Collections.Immutable.ImmutableSortedDictionary%602><br /><br /> <xref:System.Collections.Immutable.ImmutableSortedSet%601>|
+|A set for mathematical functions|<xref:System.Collections.Generic.HashSet%601><br /><br /> <xref:System.Collections.Generic.SortedSet%601>|No recommendation|<xref:System.Collections.Immutable.ImmutableHashSet%601><br /><br /> <xref:System.Collections.Immutable.ImmutableSortedSet%601>|
+
+### Collection Algorithmic Complexity
+
+| Mutable                   | Amortized  | Worst Case                | Immutable                          | Complexity |
+|---------------------------|------------|---------------------------|------------------------------------|------------|
+| `Stack<T>.Push`           | O(1)       | O(`n`)                    | `ImmutableStack<T>.Push`           | O(1)       |
+| `Queue<T>.Enqueue`        | O(1)       | O(`n`)                    | `ImmutableQueue<T>.Enqueue`        | O(1)       |
+| `List<T>.Add`             | O(1)       | O(`n`)                    | `ImmutableList<T>.Add`             | O(log `n`) |
+| `List<T>.Item[Int32]`     | O(1)       | O(1)                      | `ImmutableList<T>.Item[Int32]`     | O(log `n`) |
+| `List<T>.Enumerator`      | O(`n`)     | O(`n`)                    | `ImmutableList<T>.Enumerator`      | O(`n`)     |
+| `HashSet<T>.Add`, lookup  | O(1)       | O(`n`)                    | `ImmutableHashSet<T>.Add`          | O(log `n`) |
+| `SortedSet<T>.Add`        | O(log `n`) | O(`n`)                    | `ImmutableSortedSet<T>.Add`        | O(log `n`) |
+| `Dictionary<T>.Add`       | O(1)       | O(`n`)                    | `ImmutableDictionary<T>.Add`       | O(log `n`) |
+| `Dictionary<T>` lookup    | O(1)       | O(1) – or strictly O(`n`) | `ImmutableDictionary<T>` lookup    | O(log `n`) |
+| `SortedDictionary<T>.Add` | O(log `n`) | O(`n` log `n`)            | `ImmutableSortedDictionary<T>.Add` | O(log `n`) |
+
+A `List<T>` can be efficiently enumerated using either a `for` loop or a `foreach` loop, however; an `ImmutableList<T>` does a poor job inside a `for` loop, due to its O(log `n`) time for its indexer. It does fine when using `foreach` however. This is because `ImmutableList<T>` uses a binary tree to store its data instead of a simple array like `List<T>` uses. An array can be very quickly indexed into, whereas a binary tree must be walked down until the node with the desired index is found.
+
+Additionally, the `SortedSet<T>` has the same complexity as `ImmutableSortedSet<T>`. That's because they both use binary trees. The significant difference of course is that `ImmutableSortedSet<T>` uses an immutable one. Since `ImmutableSortedSet<T>` also offers a <xref:System.Collections.Immutable.ImmutableSortedSet%601.Builder?displayProperty=nameWithType> class that allows mutation, you can have your immutability and performance too.
```

Comment:
```suggestion
Additionally, `SortedSet<T>` has the same complexity as `ImmutableSortedSet<T>`. That's because they both use binary trees. The significant difference, of course, is that `ImmutableSortedSet<T>` uses an immutable binary tree. Since `ImmutableSortedSet<T>` also offers a <xref:System.Collections.Immutable.ImmutableSortedSet%601.Builder?displayProperty=nameWithType> class that allows mutation, you can have both immutability and performance.
```